src/layouts/RootLayout.tsx

```diff
@@ -1,24 +1,27 @@
-import { useEffect, useRef } from "react";
+import {useEffect, useState} from "react";
 import { Outlet } from "react-router-dom";
 import { setExplicitTheme, setThemeToggleState } from "../scripts/theme";
 
 export default function RootLayout(context: LiveChatContext) {
-  const toggleThemeRef = useRef<HTMLInputElement>(null);
+    /*SG Best to use useState when it's possible
+    *  useState is used to manage stateful values that are used in the rendering of a component. The state value is managed by the React framework and when it changes, React automatically re-renders the component.
+    useRef, on the other hand, is used to hold mutable values that are not directly related to the rendering of the component. The value can be updated without triggering a re-render of the component.
+    *  */
+  const [isChecked, setIsChecked] = useState<boolean>(false)
```

Comment:
Didn't think to use the "checked" property on the "input" element. Updated this functionality and also simplified the dark mode functionality.

Reply:
If you want to see the changes: https://github.com/DanutGavrus/1-app-for-a-Live-Chat-using-React-TS-Firebase-and-Tailwind/pull/19/commits/4fbc16a9471455db314fcda285585eb51c4130e5